common/src/uds_fe_programming.rs

```diff
@@ -0,0 +1,187 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    uds_fe_programming.rs
+
+Abstract:
+
+    File contains the implementation of UDS/FE programming flow that can be
+    used by both ROM and Runtime.
+
+--*/
+
+use crate::cprintln;
+use caliptra_drivers::{AxiAddr, CaliptraError, CaliptraResult, Dma, DmaOtpCtrl, SocIfc, Trng};
+
+/// UDS/FE Programming Flow
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum UdsFeProgrammingFlow {
+    /// UDS (Unique Device Secret) programming mode - 64 bytes
+    Uds,
+    /// FE (Field Entropy) programming mode - 32 bytes, can be set with a u32 granularity using bitmask
+    Fe { bitmask: u32 },
+}
+
+impl UdsFeProgrammingFlow {
+    /// Returns true if this is UDS programming mode
+    pub fn is_uds(self) -> bool {
+        matches!(self, UdsFeProgrammingFlow::Uds)
+    }
+
+    /// Returns the seed length in 32-bit words for this mode
+    pub fn seed_length(self) -> usize {
```

Comment:
nit: Can we name this function something to imply it is the number of words instead of bytes. This might be hard to read from the call site? Maybe something like `seed_num_words`.